### PR TITLE
Performace benchmarking test for s3select functions

### PR DESCRIPTION
**Pull request by grajoria**
Performance tests for s3select functions

Signed-off-by: Girjesh Rajoria <girjesh.rajoria@gmail.com>

### CODE REVIEW

File: test/s3select_perf_test.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+{
+  //std::string expected_res = "2007-01-01T00:00:00+00:00";
+  std::string input_csv;
+  size_t size = 100000;
+  generate_number_csv(input_csv, size);
+  std::string input_query = "select add(int(_1), float(_2)) from stdin;" ;
+
+  auto start = std::chrono::high_resolution_clock::now();
+
+  ProfilerStart("./add_perf.txt");
```
Comment (galsalomon66) [inline test/s3select_perf_test.cpp:35]: :+1: 

in that way, it's focused on specific flow(SQL operator)

it's also possible to make it more dynamic, i.e. not to create a TEST per operator.

File: test/s3select_perf_test.cpp [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+
+  fprintf(stderr, "Total time taken by query: %" PRIu64 " μs\nAverage time per parse per row: %" PRIu64 " μs\n",
+          elapsed_us.count(), elapsed_us.count() / (COUNT * 10));
+}
+
+TEST(TestS3SelectPerformance, like_constant)
+{
+  std::string input_csv;
+  generate_data_csv(input_csv, COUNT);
+  std::string input_query = "select count(*) from s3object where _1 like \"1_\";" ;
```
Comment (galsalomon66) [inline test/s3select_perf_test.cpp:366]: the regular expression is too short.
it is not putting enough stress on the regex-engine

File: test/s3select_perf_test.cpp [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+    ss << "9,38439,5645,2.8,Mahalia,Aldric,Mahalia.Aldric@yopmail.com,doctor,2019-04-20T20:21:22.23+05:15,__" << std::endl;
+    ss << "10,6611,7287,1.0,Pamella,Sibyls,Pamella.Sibyls@yopmail.com,police officer,2000-09-13T14:41Z,1_" << std::endl;
+  }
+  out = ss.str();
+}
+
+void generate_single_large_col_csv(std::string& out, size_t size) {
+  std::stringstream ss;
+  for (auto i = 0U; i < size; ++i) {
+    ss << "1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq" << std::endl;
```
Comment (galsalomon66) [inline test/s3select_perf_test.cpp:100]: use std::string( int , char ) instead 
for printing multiple character

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -10,5 +10,10 @@ else()
   target_link_libraries(s3select_test gtest gtest_main boost_date_time boost_thread boost_system pthread)
 endif()
 
+add_executable(s3select_perf_test s3select_perf_test.cpp)
+target_include_directories(s3select_perf_test PUBLIC ../include)
+target_link_libraries(s3select_perf_test gtest gtest_main boost_date_time boost_thread boost_system pthread)
```
Comment (galsalomon66) [inline test/CMakeLists.txt:15]: it failed the build
missing the arrow parquet libraries

**Comment by galsalomon66** [comment]
some tests had failed (rebase?)
[  FAILED  ] TestS3selectFunctions.in_expressions
[  FAILED  ] TestS3selectFunctions.predicate_as_projection_column
[  FAILED  ] TestS3selectFunctions.truefalse_in_expressions
[  FAILED  ] TestS3selectFunctions.presto_syntax_alignments

File: test/s3select_perf_test.cpp (hunk 3 above)
Comment (grajoria) [reply test/s3select_perf_test.cpp]: Done

File: test/CMakeLists.txt (hunk 4 above)
Comment (grajoria) [reply test/CMakeLists.txt]: added the libraries

File: test/CMakeLists.txt [hunk 5]
```diff
@@ -1,14 +1,20 @@
...
 
+add_executable(s3select_perf_test s3select_perf_test.cpp)
+target_include_directories(s3select_perf_test PUBLIC ../include)
+
 find_package(Arrow QUIET)
 
 if(Arrow_FOUND)
   message( "arrow is installed")
   target_link_libraries(s3select_test gtest gtest_main boost_date_time boost_thread boost_system parquet arrow pthread)
+  target_link_libraries(s3select_perf_test gtest gtest_main boost_date_time boost_thread boost_system parquet arrow pthread)
```
Comment (galsalomon66) [inline test/CMakeLists.txt:12]: :+1:

**Comment by galsalomon66** [comment]
unknown file: Failure
Unknown C++ exception thrown in the test body.
[  FAILED  ] TestS3SelectPerformance.LIKE_CONSTANT_LARGE (2070 ms)


one of the profile test is failing

**Pull request merged**